enterprise/cli/workspaceproxy.go

```diff
@@ -0,0 +1,506 @@
+		},
+		clibase.Option{
+			Flag:        "access-url",
+			Description: "Access URL of the proxy.",
+			Value:       clibase.StringOf(&proxyURL),
+		},
+		clibase.Option{
+			Flag:        "wildcard-access-url",
+			Description: "(Optional) Access url of the proxy for subdomain based apps.",
+			Value:       clibase.StringOf(&proxyWildcardHostname),
+		},
+		clibase.Option{
+			Flag:        "only-token",
+			Description: "Only print the token. This is useful for scripting.",
+			Value:       clibase.BoolOf(&onlyToken),
+		},
+	)
+	return cmd
+}
+
+type closers []func()
+
+func (c closers) Close() {
+	for _, closeF := range c {
+		closeF()
+	}
+}
+
+func (c *closers) Add(f func()) {
+	*c = append(*c, f)
+}
+
+func (r *RootCmd) proxyServer() *clibase.Cmd {
+	var (
+		cfg = new(codersdk.DeploymentValues)
+		// Filter options for only relevant ones.
+		opts = cfg.Options().Filter(codersdk.IsExternalProxies)
+
+		externalProxyOptionGroup = clibase.Group{
+			Name: "External Workspace Proxy",
+			YAML: "externalWorkspaceProxy",
+		}
+		proxySessionToken clibase.String
+		primaryAccessURL  clibase.URL
+		appSecuritYKey    clibase.String
+	)
+	opts.Add(
+		// Options only for external workspace proxies
+
+		clibase.Option{
+			Name:        "Proxy Session Token",
+			Description: "Authentication token for the workspace proxy to communicate with coderd.",
+			Flag:        "proxy-session-token",
+			Env:         "CODER_PROXY_SESSION_TOKEN",
+			YAML:        "proxySessionToken",
+			Default:     "",
+			Value:       &proxySessionToken,
+			Group:       &externalProxyOptionGroup,
+			Hidden:      false,
+		},
+
+		clibase.Option{
+			Name:        "Coderd (Primary) Access URL",
+			Description: "URL to communicate with coderd. This should match the access URL of the Coder deployment.",
+			Flag:        "primary-access-url",
+			Env:         "CODER_PRIMARY_ACCESS_URL",
+			YAML:        "primaryAccessURL",
+			Default:     "",
+			Value:       &primaryAccessURL,
+			Group:       &externalProxyOptionGroup,
+			Hidden:      false,
+		},
+
+		// TODO: Make sure this is kept secret. Idk if a flag is the best option
+		clibase.Option{
+			Name:        "App Security Key",
+			Description: "App security key used for decrypting/verifying app tokens sent from coderd.",
+			Flag:        "app-security-key",
+			Env:         "CODER_APP_SECURITY_KEY",
+			YAML:        "appSecurityKey",
+			Default:     "",
+			Value:       &appSecuritYKey,
+			Group:       &externalProxyOptionGroup,
+			Hidden:      false,
+			Annotations: clibase.Annotations{}.Mark("secret", "true"),
+		},
+	)
+
+	client := new(codersdk.Client)
+	cmd := &clibase.Cmd{
+		Use:     "server",
+		Short:   "Start a workspace proxy server",
+		Options: opts,
+		Middleware: clibase.Chain(
+			cli.WriteConfigMW(cfg),
+			cli.PrintDeprecatedOptions(),
+			clibase.RequireNArgs(0),
+			// We need a client to connect with the primary coderd instance.
+			r.InitClient(client),
```

Comment:
Shouldn't this just be a proxysdk client that we construct in the handler?

Reply:
I was honestly just using this for the version checking.